web-common/src/layout/workspace/workspace-stores.ts

```diff
@@ -40,7 +42,9 @@ class WorkspaceLayoutStore {
         parsed?.table?.height ?? DEFAULT_PREVIEW_TABLE_HEIGHT,
       );
       this.tableVisible.set(parsed?.table?.visible ?? true);
-      this.autoSave.set(parsed?.editor?.autoSave ?? true);
+      this.autoSave.set(
+        parsed?.editor?.autoSave ?? this.getAutoSaveDefault(key),
+      );
```

Comment:
Nit: Looks like you can override this. I can change `false` to `true` in localStorage and it will still auto-save. The thing that's really controlling autosave is the Editor component (for better or for worse), so I think it'd be fine to either import FILES_WITHOUT_AUTOSAVE there or use the hideAutoSave prop you added. Could leave this file as is, then.